### PR TITLE
feat(cli): add checks list/get commands with typed formatters

### PR DESCRIPTION
**Pull request by MichaelHogers**
## Summary

Add `checkly checks list` and `checkly checks get` CLI commands for viewing checks, results, and error groups from the terminal.

- REST clients for checks, check-results, check-statuses, check-groups, error-groups
- Formatters with typed field definitions (`DetailField<T>`, `ColumnDef<T>`) — fields defined once, rendered for both terminal and markdown
- Unit tests for formatters and render primitives, e2e tests for both commands

Also adds a new e2e testing account with no checks, representing a new user, to cover any potential edge cases we may miss in our own e2e prod account. 
```
config.emptyApiKey = process.env.CHECKLY_EMPTY_API_KEY
config.emptyAccountId = process.env.CHECKLY_EMPTY_ACCOUNT_ID
```

## Test plan

- [x] `vitest --run src/formatters/__tests__/ src/commands/checks/__tests__/` — 115 tests pass
- [x] E2E: `checks-list.spec.ts`, `checks-get.spec.ts` (require prod env)

### CODE REVIEW

**Comment by MichaelHogers** [comment]
checking something with totals not adding up as expected, monitors causing some funniness in the counting

File: packages/cli/src/commands/checks/get.ts [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+      default: 'detail',
+    }),
+  }
+
+  async run (): Promise<void> {
+    const { args, flags } = await this.parse(ChecksGet)
+    this.style.outputFormat = flags.output
+
+    try {
+    // Result detail mode: drill into a specific result
```
Comment (thebiglabasky) [inline packages/cli/src/commands/checks/get.ts]:
```suggestion
      // Result detail mode: drill into a specific result
```

File: packages/cli/src/formatters/__tests__/__snapshots__/check-result-detail.spec.ts.snap
Comment (thebiglabasky) [inline packages/cli/src/formatters/__tests__/__snapshots__/check-result-detail.spec.ts.snap]: That looks like a super brittle test that'll need reworking for every single edit we make to formatting. I guess it's fine, but we could likely simplify that to spot check for critical things to exist in the generated text rather than testing for the exact formatting? My assumption is that MD formatting is largely for agents anyway, so having extra tabs or return chars won't matter much.

File: packages/cli/src/rest/check-results.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  resultType: 'FINAL' | 'ATTEMPT'
+  sequenceId?: string | null
+  apiCheckResult?: ApiCheckResult | null
+  browserCheckResult?: BrowserCheckResult | null
+  multiStepCheckResult?: MultiStepCheckResult | null
+}
+
+// --- API check result ---
+
+export interface ApiCheckAssertion {
```
[the remark this answers is not in the document]
Comment (MichaelHogers) [reply packages/cli/src/rest/check-results.ts]: exactly, soon-ish

**Comment by thebiglabasky** [comment]
Note for self: came back from the SDK idea, and now sold on sticking to only zod so we can generate the openapi spec, and get runtime validation all in a single place

File: packages/cli/src/commands/checks/get.ts [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        }).catch(() => ({ data: { entries: [], nextId: null, length: 0 } })),
+        api.errorGroups.getByCheckId(args.id).catch(() => ({ data: [] })),
+      ])
+
+      const status = statusResp.data
+      const { entries: results, nextId } = resultsResp.data
+      const errorGroups = errorGroupsResp.data
+
+      if (flags.output === 'json') {
+        this.log(JSON.stringify({ check, status, results, nextId, errorGroups }, null, 2))
```
[the remark this answers is not in the document]
Comment (MichaelHogers) [reply packages/cli/src/commands/checks/get.ts]: will leave it for now, consistent with other CLIs, e.g. gh api, gh pr view --json commands

**Review by thebiglabasky — APPROVED** [review]
LGTM!
